geoviews/plotting/mpl/chart.py

```diff
@@ -0,0 +1,156 @@
+    If the magnitude is small and only needs a single half-line and no full lines or
+    triangles, the half-line is offset from the end of the barb so that it can be
+    easily distinguished from barbs with a single full line. The magnitude for the barb
+    shown above would nominally be 65, using the standard increments of 50, 10, and 5.
+    """
+
+    padding = param.ClassSelector(default=0.05, class_=(int, float, tuple))
+
+    from_uv_components = param.Boolean(default=False, doc="""
+        If True, the vdims are expected to be in the form of
+        wind components (U and V); else, wind direction and speed.""")
+
+    style_opts = [
+        "alpha",
+        "color",
+        "edgecolors",
+        "facecolors",
+        "linewidth",
+        "marker",
+        "visible",
+        "cmap",
+        "norm",
+        # barb specific
+        "length",
+        "barbcolor",
+        "flagcolor",
+        "fill_empty",
+        "rounding",
+        "barb_increments",
+        "flip_barb",
+        "pivot",
+        "sizes",
+        "width",
+    ]
+
+    _nonvectorized_styles = [
+        "alpha",
+        "marker",
+        "cmap",
+        "visible",
+        "norm",
+        # TODO: clarify whether these are vectorized or not
+        "length",
+        "barbcolor",
+        "flagcolor",
+        "fill_empty",
+        "rounding",
+        "barb_increments",
+        "flip_barb",
+        "pivot",
+        "sizes",
+        "width",
+    ]
+
+    _plot_methods = dict(single="barbs")
+
+    def _get_us_vs(self, element):
+        if self.from_uv_components:
+            us = element.dimension_values(2, flat=False) if len(element.data) else []
+            vs = element.dimension_values(3, flat=False) if len(element.data) else []
+            uv_magnitudes = np.hypot(us, vs)  # unscaled
+            radians = (np.pi / 2.0) - np.arctan2(us / uv_magnitudes, vs / uv_magnitudes)
+            element = element.add_dimension("Angle", 4, radians, vdim=True)
+            element = element.add_dimension("Magnitude", 5, uv_magnitudes, vdim=True)
+            mag_dim = element.get_dimension(5)
+        else:
+            radians = element.dimension_values(2) if len(element.data) else []
+            mag_dim = element.get_dimension(3)
+
+        if isinstance(mag_dim, str):
+            mag_dim = element.get_dimension(mag_dim)
+
+        if isinstance(mag_dim, dim):
+            magnitudes = mag_dim.apply(element, flat=True)
+        else:
+            magnitudes = element.dimension_values(mag_dim)
+
+        if self.invert_axes:
+            radians += 1.5 * np.pi
+
+        # Compute U, V to serialize to matplotlib
+        # Even if it's from U, V components, we still need to re-compute
+        # because operations may have been applied to magnitudes
+        if not self.from_uv_components or isinstance(mag_dim, dim):
+            us = magnitudes * np.cos(radians.flatten())
```

Comment:
As I read the code `radians` from `from_uv_components` is not in cartesian coordinates but in north-clockwise coordinates (or a better. Is this conversion still the same for this case?

Reply:
I think I fixed it. To disregard reasoning thru the trig, I just compared it against metpy + matplotlib which I believe should be good enough (but let me know if it isn't).
https://github.com/holoviz/geoviews/pull/651#issuecomment-1610239857